src/tests/ftest/launch.py

```diff
@@ -52,2294 +68,2973 @@
-                find_char = " "
-            exe_name = cmd[0:cmd.find(find_char)]
-        except RuntimeError:
-            exe_name = None
-
-        if exe_name:
-            cmd = [
-                "gdb", "-cd={}".format(daos_cores_dir),
-                "-ex", "set pagination off",
-                "-ex", "thread apply all bt full",
-                "-ex", "detach",
-                "-ex", "quit",
-                exe_name, corefile
-            ]
-            stack_trace_file = os.path.join(
-                daos_cores_dir, "{}.stacktrace".format(corefile))
-            try:
-                with open(stack_trace_file, "w") as stack_trace:
-                    stack_trace.writelines(get_output(cmd, check=False))
-            except IOError as error:
-                log("Error writing {}: {}".format(stack_trace_file, error))
-                return_status = False
-            except RuntimeError as error:
-                log("Error creating {}: {}".format(stack_trace_file, error))
-                return_status = False
-        else:
-            log(
-                "Unable to determine executable name from gdb output: '{}'\n"
-                "Not creating stacktrace".format(gdb_output))
-            return_status = False
-        log("Removing {}".format(corefile_fqpn))
-        os.unlink(corefile_fqpn)
+        if "core files" in summary:
+            # Process the core files
+            return_code |= self._process_core_files(os.path.split(destination)[0])
 
-    return return_status
+        return return_code
 
+    def _list_files(self, hosts, source, pattern, depth):
+        """List the files in source with that match the pattern.
 
-def get_test_category(test_file):
-    """Get a category for the specified test using its path and name.
+        Args:
+            hosts (NodSet): hosts on which the files are located
+            source (str): where the files are currently located
+            pattern (str): pattern used to limit which files are processed
+            depth (int): max depth for find command
 
-    Args:
-        test_file (str): the test python file
+        Returns:
+            tuple: a tuple containing:
+                int: 0 = success, 16 = failure
+                NodeSet: hosts with at least one file matching the pattern in the source directory
 
-    Returns:
-        str: concatenation of the test path and base filename joined by dashes
+        """
+        status = 0
+        hosts_with_files = NodeSet()
+        source_files = os.path.join(source, pattern)
+        logger.debug("-" * 80)
+        logger.debug("Listing any %s files on %s", source_files, hosts)
+        other = ["-printf", "'%M %n %-12u %-12g %12k %t %p\n'"]
+        result = run_remote(logger, hosts, find_command(source, pattern, depth, other))
+        if not result.passed:
+            message = f"Error determining if {source_files} files exist on {hosts}"
+            self._fail_test(self.result.tests[-1], "Process", message)
```

Comment:
(pylint-no-value-for-parameter) No value for argument 'fail_reason' in staticmethod call